### PR TITLE
efont-unicode: init at 0.4.2

### PR DESCRIPTION
**Pull request by ncfavier**
###### Motivation for this change

Adds the [/efont/ Unicode](http://openlab.ring.gr.jp/efont/unicode/) bitmap font, with `.pcf.gz` and `.otb` files in the `out` output and `.bdf` files in the `bdf` output.

Other bitmap font packages I looked at (`siji`, `tewi-font`, `dina-font`) seem to be called with
```nix
{ inherit (buildPackages.xorg) mkfontscale fonttosfnt; }
```
in `all-packages.nix`, what is that for? Should I do that as well? Why isn't `bdftopcf` included there?

cc @rnhmjoj 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126593` at a1b1d90b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>efont-unicode</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/data/fonts/efont-unicode/default.nix:16:3:
```
   |
16 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/data/fonts/efont-unicode/default.nix:30:3:
```
   |
30 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by rnhmjoj** [comment]
IIRC the `buildPackages` thing is to fix cross-compilation. If you try to build your package under `pkgsCross.someplatform` instead of `pkgs` you should see a failure.

**Comment by ncfavier** [comment]
`nix-build -A pkgsCross.raspberryPi.efont-unicode` builds fine. Aren't cross-compilation issues handled by `buildInputs`, `nativeBuildInputs` and consorts?

**Comment by rnhmjoj** [comment]
`buildPackages` is needed because the cross-compilation infrastructure can't deal with overrides. For example, if a package is built with `callPackage ./something { boost = boost_166; }`, it won't automagically cross-compile under the `pkgsCross` overlay. Instead `callPackage ./something { boost = buildPackages.boost_166; }` will.

Now, some fonts use an "override" `{ inherit (xorg) fonttosfnt mkfontdir; }`, this should break as well under `pkgsCross`, the solution again is to use `buildPackges.xorg`. Your package is working because it has no such override, still I don't understand how callPackage is picking up `mkfontscale` and `fonttosfnt`, since these do not seem to be in the top-level, or are they?

**Comment by ncfavier** [comment]
Seems like `callPackage` makes the `xorg` package set available at the top-level:
https://github.com/NixOS/nixpkgs/blob/2c31bc7f200b6a6db59787bd6ac6f9d48d69ed89/pkgs/top-level/splice.nix#L125-L127

**Comment by rnhmjoj** [comment]
Right, well, it looks a bit arbitrary (and confusing) to pull in xorg to me. I'd prefer the explicit passing of the xorg attribute (even if it requires the `buildPackages`), but it's ok.

**Comment by ncfavier** [comment]
Yeah I'm a bit confused too.

Would specifying only `xorg` in the arguments and then using `xorg.fonttosfnt` etc. be better? That doesn't list the dependencies explicitly in the arguments but I don't know if that's a real problem since they're already listed in `nativeBuildInputs`.

**Comment by rnhmjoj** [comment]
> Would specifying only xorg in the arguments and then using xorg.fonttosfnt etc. be better? 

It seems there is no significant difference between the two. Based on the number of packages in Nixpkgs, directly pulling in `xorg` seems the preferred solution.

**Comment by ncfavier** [comment]
Done.

**Review by rnhmjoj — CHANGES_REQUESTED** [review]
I tested a few files and the fonts look ok (both opentype and X11). The package looks good and seems deterministic. Can you just fix this small issue?

File: pkgs/data/fonts/efont-unicode/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    # convert bdf fonts to otb
+    for f in *.bdf; do
+        faketime -f "1970-01-01 00:00:01" \
+        fonttosfnt -v -m 2 -o "''${f%.bdf}.otb" "$f"
+    done
+  '';
+
+  installPhase = ''
+    dir=share/fonts/misc
```
[the remark this answers is not in the document]
Comment (ncfavier) [reply pkgs/data/fonts/efont-unicode/default.nix:36]: Done!

**Pull request merged**